datafusion/functions-aggregate/src/min_max.rs

```diff
@@ -645,19 +645,29 @@ fn min_max_batch_struct(array: &ArrayRef, ordering: Ordering) -> Result<ScalarVa
             }
         }
     }
-    // use force_clone to free array reference
-    Ok(extreme.force_clone())
+
+    Ok(extreme)
 }
 
 macro_rules! min_max_struct {
     ($VALUE:expr, $DELTA:expr, $OP:ident) => {{
         if $VALUE.is_null() {
-            $DELTA.clone()
+            let mut delta_copy = $DELTA.clone();
+            // When the new value won we want to compact it to
+            // avoid storing the entire input
+            delta_copy.compact();
```

Comment:
I moved compaction here so it's only done if we're definitely going to keep a reference to the new value.

Reply:
it would be great to try and avoid the duplication in these min/max macros anyways (and instead use an Array)  -- no changes are neede din this PR, I am just observing